cloud/services/securitygroups/securitygroups.go

```diff
@@ -113,6 +126,45 @@ func (s *Service) Reconcile(ctx context.Context, spec interface{}) error {
 	return err
 }
 
+func ruleExists(rules []network.SecurityRule, rule controlPlaneDefaultRule) bool {
+	for _, existingRule := range rules {
```

Comment:
nit: for better code performance, we could put the rules in a map[string]network.SecurityRule before passing them into the ruleExists function, that way we don't have to loop through the rules for each default rules and the cost is O(1) instead of O(n). This isn't a very big deal here because we only have 2 default rules so the cost is pretty small anyways but using a map would scale better.